File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/repositories/CheckResultRepository.java
package io.github.sanyavertolet.edukate.backend.repositories;

import io.github.sanyavertolet.edukate.backend.entities.CheckResult;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Flux;

@Repository
public interface CheckResultRepository extends ReactiveMongoRepository<CheckResult, String> {
    Flux<CheckResult> findBySubmissionId(String submissionId);
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/repositories/CheckResultRepository.kt
package io.github.sanyavertolet.edukate.backend.repositories

import io.github.sanyavertolet.edukate.backend.entities.CheckResult
import org.springframework.data.domain.Sort
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import org.springframework.stereotype.Repository
import reactor.core.publisher.Flux

@Repository
interface CheckResultRepository : ReactiveCrudRepository<CheckResult, Long> {
    fun findBySubmissionId(submissionId: Long, sort: Sort): Flux<CheckResult>
}
